.gitlab-ci.yml

```diff
@@ -0,0 +1,64 @@
+stages:
+  - test
+  - build
+
+image:                             parity/rust:nightly
+
+variables:
+  CI_SERVER_NAME:                  "GitLab CI"
+  CARGO_HOME:                      "${CI_PROJECT_DIR}/.cargo"
+
+
+cache:
+  key:                             "${CI_JOB_NAME}"
+  paths:
+    - ./target/
+    - ./.cargo/
+
+
+.collect_artifacts:                &collect_artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     7 days
+    paths:
+    - target/release/shasper
+
+.determine_version: &determine_version |
+  export VERSION="$(sed -r -n '1,/^version/s/^version = "([^"]+)".*$/\1/p' Cargo.toml)"
+  echo "Version" $VERSION
+
+before_script:
+  - ./scripts/init.sh
+  - export PATH="${CI_PROJECT_DIR}/cargo/bin/:$PATH"
+  - ./scripts/build.sh
+
+
+#### stage:                        test
+
+test:rust:stable:
+  stage:                           test
+  script:
+    - time cargo test --all --locked
```

Comment:
Sorry about the confusion on this, but I rethink about this and we probably still need `--release` parameters here. The issue is that when we get any integration tests, debug build tests might start to be slow.